### PR TITLE
Fix VertxGrpcExporter reponse status handling

### PR DESCRIPTION
**Pull request by luneo7**
GRPC status and status message comes from trailers so it needs to be handled in the end handler, if doing before the message if fully sent, trailers are not there yet so there won't be no GRPC status and also no status message.

### CODE REVIEW

**Comment by brunobat** [comment]
Thanks @luneo7. Will checkout the PR and see what happens with the demo apps.

**Review by brunobat — COMMENTED** [review]
Do you think you can come up with a test for this @luneo7?
I think this makes total sense. 
@alesj, what do you think?

File: extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/VertxGrpcExporter.java [hunk 1]
```diff
@@ -201,20 +202,39 @@ public void handle(GrpcClientRequest<Buffer, Buffer> request) {
...
-                        }
+                        response.exceptionHandler(new Handler<>() {
+                            @Override
+                            public void handle(Throwable t) {
+                                exporterMetrics.addFailed(numItems);
+                                logger.log(
+                                        Level.SEVERE,
+                                        "Failed to export "
+                                                + type
+                                                + "s. The response could not be parsed. Full error message: "
```
Comment (alesj) [inline extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/VertxGrpcExporter.java:213]: How do you know that the response could not be parsed?

File: extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/VertxGrpcExporter.java [hunk 2]
```diff
@@ -201,20 +202,39 @@ public void handle(GrpcClientRequest<Buffer, Buffer> request) {
...
+                                handleError(error.status, response);
+                            }
+                        }).endHandler(new Handler<>() {
+                            @Override
+                            public void handle(Void ignored) {
+                                GrpcStatus status = getStatus(response);
+                                if (status == GrpcStatus.OK) {
+                                    exporterMetrics.addSuccess(numItems);
+                                    result.succeed();
+                                    return;
```
Comment (alesj) [inline extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/VertxGrpcExporter.java:229]: Perhaps rather do "else"

**Comment by alesj** [comment]
Looks good otherwise, with a few cosmetic fixes.
But yeah, a test would be nice / good, if not even mandatory. :-)

**Comment by Sgitario** [comment]
Any updates about this pull request? We're experiencing the same issue and confirmed that these changes fixed it.

**Comment by brunobat** [comment]
@luneo7 will you complete this PR?

**Comment by dastrobu** [comment]
Would be great if we can see a fix for this issue soon.

**Comment by luneo7** [comment]
Will work on this tomorrow, and will check if there is a way to do some testing, as to test this is somehow difficult (it even doesn't have any test right now)...

File: extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/VertxGrpcExporter.java (hunk 1 above)
Comment (luneo7) [reply extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/VertxGrpcExporter.java]: Right... actually the stream failed, changed the error log to show that.

**Comment by luneo7** [comment]
Checking the existing tests, and also checking OTEL repos, our tests are identical to theirs, even the exporters integration test (https://github.com/open-telemetry/opentelemetry-java/blob/695ed5350b2ef218c38087a627549f4de7b86c22/integration-tests/otlp/src/main/java/io/opentelemetry/integrationtest/OtlpExporterIntegrationTest.java does the same thing that `integration-tests/opentelemetry-vertx-exporter` does). The tests right now get that the implementation is fully working. The thing before it wasn't handling the streaming response correctly, I couldn't figure out an IT to reproduce the original issue, although it is fixed hahaha.

I've addressed the comments, though... if you want to go forward without a new test...

**Review by brunobat — APPROVED** [review]
Thanks very much @luneo7.
Works for me.

**Comment by brunobat** [comment]
Should fix: https://github.com/quarkusio/quarkus/issues/35686

**Comment by famod** [comment]
Nice, will try this ASAP.

If it fixes the issues I'd strongly suggest backporting it.

**Comment by famod** [comment]
This fix should also take effect if cherry-picked on top of the 3.7 branch, shouldn't it?
Thats what I did and I'm still seeing the issue described in #35686. :-/

**Pull request merged**